File: 1-two-sum/two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mpp = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (mpp.containsKey(complement)) {
                return new int[] { mpp.get(complement), i };
            }

            mpp.put(nums[i], i);
        }

        return new int[] { -1, -1 };  // No valid pair found
    }
}

File: 1-two-sum/two-sum.cpp
#include <unordered_map>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> prevMap;

        for(int i = 0; i < nums.size(); i++) {
            int currentNum = nums[i];
            int complement = target - currentNum;
            if(prevMap.count(complement)) {
                return {prevMap[complement], i};
            }
            prevMap[currentNum] = i;
        }
        return {};
    }
};